### ISSUE TITLE
when 'Publish Project' is chosen, don't ask the user if he/she wants to save unsaved documents

### ISSUE THREAD

**Issue by ginobean** [spritebuilder/SpriteBuilder#1262]
If you look at Xcode as an example, Xcode always assumes that you want to save your changes. So if you click on 'R'un with unsaved changes, I believe it will automatically save your current changes and then do the actual compile and run. 

The problem with this dialog is that it seems intended to handle a case that I have never run into myself -- where I have changes that I've just made, but I don't want to publish them ? So this is adding unnecessary friction to the publishing process. This was probably an issue with CocosBuilder as well. Of the hundreds of times that I've published a project using either CocosBuilder or SpriteBuilder, the number of times where I have _not_ wanted to save my unsaved changes has been exactly _zero_.

I think you can safely assume that the user always wants to save any/all unsaved changes as part of the Publish Project action. At the very least, make this the _default_ and if some users want that confirm dialog, they can explicitly set it themselves..

**Comment by LearnCocos2D** [comment]
Good point. IDEs often also ask to save source files just before compiling. They all have a Preferences option to turn this off and "always save modified files". SB should have that setting since it only makes sense not to save in few instances, and typically it's only just a matter of (possibly) speeding up the publish by a short amount of time.
